CRM/Contact/BAO/Relationship.php

```diff
@@ -1452,6 +1455,20 @@ public static function relatedMemberships($contactId, &$params, $ids, $action =
       }
     }
 
+    // CRM-15829 UPDATES
+    // If we're looking for active memberships we must consider pending (id: 5) ones too.
+    // Hence we can't just call CRM_Member_BAO_Membership::getValues below with the active flag, is it would completely miss pending relatioships.
+
+    $query = 'SELECT * FROM `civicrm_membership_status`';
+    if ($active) {
+      $query .= 'WHERE `is_current_member` = 1 OR `id` = 5';
```

Comment:
Please don't use constant '5' for 'Pending' as membership_status.id values are not the same for all sites.
`$pendingStatusId = array_search('Pending', CRM_Member_PseudoConstant::membershipStatus());`